File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        int n = nums.size();
        vector<int> ans(n, 0);
        int curr = nums[0]; ans[1] = curr;
        for(int i = 1; i < nums.size()-1; i++) {
            curr *= nums[i];
            ans[i+1] = curr;
        }
        curr = nums[nums.size()-1]; ans[nums.size()-2]*=curr;
        for(int i = nums.size()-2; i > 0; --i) {
            curr *= nums[i];
            ans[i-1] *= curr;
        }
        ans[0] = curr;
        return ans;
    }
};

File: Data Structures & Algorithms/reverse-a-linked-list/submission-0.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next

class Solution:
    def reverseList(self, head: Optional[ListNode]) -> Optional[ListNode]:
        trail = head
        if trail == None:
            return None
        curr = head.next
        trail.next = None
        while curr != None:
            ahead = curr.next
            curr.next = trail
            trail = curr
            curr = ahead
        return trail
        